### PR TITLE
Justification ranking

### PR DESCRIPTION
**Pull request by JoshAddington**
Ranking of justification was initially implemented in the GUI, but didn't persist if the card modal was closed and opened again. 

Ranking fields have been added to both the Django and Backbone models, and sorting has been added to Backbone models and views.

### CODE REVIEW

File: project/static/js/cards/views.js [hunk 1]
```diff
@@ -358,6 +375,7 @@
           fail = function(){
             alert('an error occurred; please try to re-save');
           };
+      console.log(patch);
```
Comment (shapiromatron) [inline project/static/js/cards/views.js]: remove logging

File: project/static/js/cards/models.js [hunk 2]
```diff
@@ -121,6 +126,22 @@
...
+      var factors = this.getFactors();
+      var rankedFactors = _.filter(factors, function(d){
+        return this.get(d) !== null;
+      }, this);
+      var factorsWithRanks = _.map(rankedFactors, function(d){
+          return {factor: d, rank: this.get(d)};
+      }, this);
+      var sortedFactors = _.sortBy(factorsWithRanks, 'rank');
+      return sortedFactors;
+    },
```
Comment (shapiromatron) [inline project/static/js/cards/models.js]: I think we can chain these events using underscore (not tested): 

```javascript
return _.chain(this.getFactors())
  .filter(function(d){return this.get(d) !== null;})
  .map(function(d){return {factor: d, rank: this.get(d);}, this)
  .sortBy('rank')
  .value();
```

File: project/cards/models.py [hunk 3]
```diff
@@ -43,9 +73,14 @@ class BoardCards(models.Model):
         default=0)
     last_updated = models.DateTimeField(auto_now=True)
 
+    def ranked_placement(self):
```
Comment (shapiromatron) [inline project/cards/models.py]: change to a `@property` and make the name `placement_ranks_completed`?

File: project/static/js/cards/views.js [hunk 4]
```diff
@@ -307,21 +308,21 @@
...
           updateSelections = function() {
-            $ordered.find('.rank').each(function(i, d) {
-              return $(d).attr('value', i + 1).text(" (#" + (i + 1) + ")");
-            });
             $unordered.find('.rank').each(function(i, d) {
+              $(d).removeAttr('value');
               return $(d).text("");
             });
+            $ordered.find('.rank').each(function(i, d) {
+              return $(d).attr('value', i + 1).text(" (#" + (i + 1) + ")");
```
Comment (shapiromatron) [inline project/static/js/cards/views.js]: If in the ordered list, can we set the checkbox as checked as well?

Comment (JoshAddington) [inline project/static/js/cards/views.js]: Yep. Did the same to unordered.find

**Comment by shapiromatron** [comment]
@JoshAddington Nice job!

**Pull request merged**